File: EmojiPanel/src/main/java/com/binlee/emoji/ImageAdapter.java
package com.binlee.emoji;

import com.binlee.emoji.third.GlideImageEngine;
import com.binlee.emoji.compat.ImageEngine;

public final class ImageAdapter {

    private static ImageEngine sEngine;

    private ImageAdapter() {
        throw new AssertionError("no instance.");
    }

    public static void setImageEngine(ImageEngine engine) {
        if (engine == null) {
            throw new NullPointerException("setImageEngine(): engine is null");
        }
        sEngine = engine;
    }

    public static ImageEngine engine() {
        if (sEngine == null) {
            sEngine = new GlideImageEngine();
        }
        return sEngine;
    }
}

File: EmojiPanel/src/main/java/com/binlee/emoji/ImageAdapter.kt
package com.binlee.emoji

import com.binlee.emoji.compat.ImageEngine
import com.binlee.emoji.third.GlideImageEngine


class ImageAdapter private constructor() {
    companion object {
        private var sEngine: ImageEngine? = null
        fun setImageEngine(engine: ImageEngine?) {
            if (engine == null) {
                throw NullPointerException("setImageEngine(): engine is null")
            }
            sEngine = engine
        }

        fun engine(): ImageEngine? {
            if (sEngine == null) {
                sEngine = GlideImageEngine()
            }
            return sEngine
        }
    }

    init {
        throw AssertionError("no instance.")
    }
}